### ISSUE TITLE
() is not the empty type

### ISSUE THREAD

**Issue by BartMassey** [stevedonovan/gentle-intro#59]
In 1-basics: "The () type is the empty type, nada, void, zilch, nothing. Everything in Rust has a value, but sometimes it's just nothing." The type `()` actually is a type with a single value, `()`. It is usually referred to as the "unit type", since it is inhabited by just the one value. The empty type, the type inhabited by no values, is `!`.

This is important, because the unit value is useful in programs:

```Rust
match v {
    Some(v) => println!("{}", v),
    None => (),
}
```

**Comment by stevedonovan** [comment]
Yes, that was sloppy writing - it is definitely not _nothing_.  I think I was still confused about `void` in those days.
